Documentation/swagger-extensions.md

```diff
@@ -0,0 +1,180 @@
+# AutoRest Extensions for Swagger 2.0
+
+## Introduction
+The following documents describes AutoRest specific vendor extensions for [Swagger 2.0](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md) schema. Some of the extensions are only applicable to Microsoft Azure and as such are only available in Azure code generators (e.g. Azure.CSharp, Azure.NodeJS, etc.).
+
+## Generic Extensions
+* [x-ms-skip-url-encoding](#x-ms-skip-url-encoding) - skips URL encoding for path and query parameters
+* [x-ms-enum](#x-ms-enum) - additional metadata for enums
+* [x-ms-parameter-grouping](#x-ms-parameter-grouping) - groups method parameters in generated clients
+* [x-ms-paths](#x-ms-paths) - alternative to [Paths Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#pathsObject) that allows [Path Item Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#pathItemObject) to have query parameters for non pure REST APIs
+* x-ms-client-name - *not currently implemented*
+* [x-ms-external](#x-ms-external) - allows specific [Definition Objects](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#definitionsObject) to not be excluded from code generation
+
+## Microsoft Azure Extensions
+* x-ms-odata
+* x-ms-pageable
+* x-ms-long-running-operation
+* x-ms-azure-resource
+* x-ms-discriminator-value 
```

Comment:
Custom polymorphic discriminators have value outside Azure-specific scenarios. I'm pretty sure I implemented this extension in the base (non-Azure) code generators.